### PR TITLE
Disable Swap form while loading best price

### PR DESCRIPTION
**Pull request by shoom3301**
# Summary

Context: https://cowservices.slack.com/archives/C036G0J90BU/p1680169031535319

Basically, there's some cases where we allow users to submit orders with a "fast" price from the quote endpoint. 
In this PR we are displaying "Hang in there. Calculating best price..." state until we get the best price.

image

  # To Test

1. Open vercel from https://github.com/cowprotocol/cowswap/pull/2255
2. Open swap page, setup a trade, run `localStorage.setItem('delayBestQuote', '200000')` in console and reload the page
3. Keep developed tools open in network tab
4. Just after page load do a swap
- [ ] check POST /orders request, it should NOT contain`quoteId` field
5. Open vercel from the current PR
6. Open swap page, setup a trade, run `localStorage.setItem('delayBestQuote', '10000')` in console and reload the page
- [ ] first ~10 seconds "Hang in there. Calculating best price..." must be displayed
- [ ] first ~10 seconds it's impossible to swap
7. Do a swap when it's possible
- [ ] check POST /orders request, it MUST contain`quoteId` field with a number value
8. After all tests run `localStorage.removeItem('delayBestQuote')` in console

image

### CODE REVIEW

**Comment by shoom3301** [comment]
@elena-zh thanks!
1. Fixed
2. Out of scope
3. Out of scope
4. Out of scope
5. Out of scope
6. It's ok

**Comment by elena-zh** [comment]
Hey @shoom3301 m case 1 LGTM!
Also, for the case 4 I noticed that `localStorage.setItem('delayBestQuote', '10000')` affects this behavior. But if is going to be removed before going live, it should not affect fees displaying. 

As for the case 5, I have opened #2258 issue for this. 

Thanks.

**Review by elena-zh — APPROVED** [review]

**Review by anxolin — APPROVED** [review]
It's simple and nice. The only small thing is, that in the original i one reproducing this case, was making sure that we throw an error if this ever happens again. 

Basically the signAndPost function was throwing if the quote is not final. 


Some other ubernit, is that in principle we need to block the user in their last step only. Basically when they:
- Press post order button in the confirmation modal (normal flow)
- Press post order button in the widget (expert mode)


Both comments above are nitpicks, specially if we are SURE its impossible to post orders from now on with "fast" prices

File: src/custom/hooks/useRefetchPriceCallback.tsx [hunk 1]
```diff
@@ -252,7 +252,20 @@ export function useRefetchQuoteCallback() {
...
+
+          // TODO: remove after testing
+          // The flag serves to make an artificial delay before getting "best" quote
+          if (bestQuoteDelay) {
+            setTimeout(() => {
+              handleResponse(res, true)
+            }, +bestQuoteDelay)
+            return
+          }
+
```
Comment (alfetopito) [inline src/custom/hooks/useRefetchPriceCallback.tsx:266]: Don't forget to remove this!!

(Happened to me before)

**Review by alfetopito — APPROVED** [review]

**Pull request merged**